src/components/Menu/index.tsx

```diff
@@ -22,102 +29,125 @@ type MenuProps = {
+    const combinedDates = [state.files['parentPageLastUpdatedDate']];
+
+    if (state.files[props.filterKey]) {
+      combinedDates.push(...state.files[props.filterKey]);
+    } else if (state.files['all'] && state.files['all'].length > 0) {
+      combinedDates.push(...state.files['all']);
     }
-    if (this.state.isOpen) {
-      return (
-        <MenuStyle>
+
+    lastUpdatedDate = Math.max(
+      ...combinedDates.map((e) => new Date(e).getTime())
+    );
+  }
+
+  if (isOpen) {
+    return (
+      <MenuStyle>
+        <div>
           <div>
-            <div>
-              <MenuHeaderStyle>
-                <MenuCloseButton closeMenu={this.closeMenu} />
-                {typeof this.props.filterKey !== 'undefined' && (
-                  <FilterSelect
-                    filters={this.props.filters}
-                    filterKey={this.props.filterKey}
-                    filterKind={this.props.filterKind}
-                    url={this.props.url}
-                  />
-                )}
-              </MenuHeaderStyle>
-              <MenuBodyStyle>
-                {showVersionSwitcher && (
-                  <VersionSwitcher url={this.props.url} />
-                )}
-                {showLibVersionSwitcher && (
-                  <LibVersionSwitcher url={this.props.url} />
-                )}
-                <Directory
-                  filterKey={this.props.filterKey}
-                  url={this.props.url}
-                />
-                <MenuBreakStyle />
-                <RepoActions
-                  url={this.props.url}
-                  directoryPath={this.props.directoryPath}
+            <MenuHeaderStyle>
+              <MenuCloseButton closeMenu={closeMenu} />
+              {typeof props.filterKey !== 'undefined' && (
+                <FilterSelect
+                  filters={props.filters}
+                  filterKey={props.filterKey}
+                  filterKind={props.filterKind}
+                  url={props.url}
                 />
-              </MenuBodyStyle>
-            </div>
+              )}
+            </MenuHeaderStyle>
+            <MenuBodyStyle>
+              {showVersionSwitcher && <VersionSwitcher url={props.url} />}
+              {showLibVersionSwitcher && <LibVersionSwitcher url={props.url} />}
+              <Directory filterKey={props.filterKey} url={props.url} />
+              <MenuBreakStyle />
+              <RepoActions
+                url={props.url}
+                directoryPath={props.directoryPath}
+              />
+              <LastUpdatedStyle>
+                {displayLastUpdatedString(lastUpdatedDate)}
+              </LastUpdatedStyle>
+            </MenuBodyStyle>
           </div>
-        </MenuStyle>
-      );
-    }
-    return <MenuOpenButton openMenu={this.openMenu} />;
+        </div>
+      </MenuStyle>
+    );
+  }
+  return <MenuOpenButton openMenu={openMenu} />;
+}
+
+export default forwardRef(Menu);
```

Comment:
Have to use `forwardRef` since the `ref` property being used when the `Menu` component is created inside the `Page` component.

The `ref` is used so that the `SidebarLayoutToggle` component can open and close the menu on mobile screen sizes